src/main/java/com/limechain/network/protocol/warp/DigestHelper.java

```diff
@@ -37,6 +40,14 @@ private void updateEpochStateIfBabeConsensusMessageExists(HeaderDigest[] headerD
                 .findFirst()
                 .map(HeaderDigest::getMessage)
                 .ifPresent(message -> epochState.updateNextEpochBlockConfig(message));
+
+        if (BlockState.getInstance().isFullSyncFinished()) {
```

Comment:
This field is removed on my branch. Could you give me more info on why you need this here so I know when I am refactoring?

Reply:
This runtime call isn't accessible in the earlier runtime state. The best solution I've found is to wait until the full sync finishes, at this point, the runtime can be called with BabeApi_current_epoch. I had two options for implementing the logic related to updating the current epoch fields: DigestHelper or BlockState. From a data encapsulation perspective, placing the logic in BlockState would be preferable since it keeps fullSyncFinished without needing a getter. However, the functionality seems more relevant to DigestHelper than BlockState, so I decided to add a getter and place the code within DigestHelper instead.